### ISSUE TITLE
gtp2ogs restart with many in-flight games can cause instability

### ISSUE THREAD

**Issue by windo** [online-go/gtp2ogs#92]
This one might again be more about my suboptimal bot setup, but with ~10 concurrent games, restarting `gtp2ogs` was very problematic. My bot takes several seconds to initialize the neural network on startup and with many games, it was taking longer. I don't know what exactly was going on, but I think it was taking long enought that either the web sockets were timing out or the bot pipes were timing out - in any case I was having a hard time getting the bot to connect to all the games again.

I don't know exactly what the solution should be. I think `gtp2ogs` could probably rate limit the starting of new bots or maybe just the socket timeouts can be increased.

**Comment by wonderingabout** [comment]
i use `--maxtotalgames` and `--maxactivegames` due to limited GPU power, but i tried many games at a time without having any issue in the past with phoenixgo

**Comment by wonderingabout** [comment]
i'll try this later on beta @windo

**Comment by windo** [comment]
I had this trouble with `amybot-beginner` and it currently runs with 10 visits (on CPU, not even GPU) so each move takes very little resources. It was just that at startup initializing 10 (or so) sets of engines was taking a so long as to time out the games and `gtp2ogs` seemed not to try recover from that state but just left the games that had failed hanging.

**Comment by wonderingabout** [comment]
i understand, so not a hardware issue but software

i'll see if i have this issue witth phoenixgo on beta and tell you the results

also, come chat with us on discord :) https://discord.gg/HZ23Cp9 in #bots channel

**Comment by windo** [comment]
I will try to add better logging for my gtp2ogs processes and will report back if I figure out the circumstances where this happens for my system.

**Comment by windo** [comment]
I haven't had this problem since I started using corrqueue - it serializes the bot startups for a good chunk of games. I've also started running the bots on a bigger systems (i was running them on a tiny NUC computer before). I also expect that even if some problem still exists here, the other tweaks to handle hung games and bot crashes should help smooth things over.

So I'd be inclined to close this for now, maybe after #131 is merged (in case part of the problem was that some of the bot processes crashed due to all of them trying to initialize at the same time).
